### ISSUE TITLE
The Infinite Fight Scene

### ISSUE THREAD

**Issue by filiph** [NaNoGenMo/2017#127]
This is a hack of my [IFCOMP entry](https://egamebook.com/vermin) (a procedural gamebook) that removes all interactivity, switches from second person to third person, and keeps the initial fight going ad infinitum.

Here's the code: https://github.com/filiph/edgehead/tree/nanogenmo2017

You run it by cloning the repo, installing Dart, running `pub get` and then `dart bin/play.dart`. It's actually very slow, partly because the actors in the book are planning each move, and partly because this is _not_ what this thing is optimized for.

I'll add an output soon. Here's an excerpt from the middle of the novel:

> Briana takes the scimitar and the sword. From the blackness of the caverns, another goblin comes into view. He lifts his fist.
> 
> Briana casts her spear at the goblin. The spear drills through the goblin's shoulder. The goblin grunts in pain and springs at Briana.
> 
> Briana dodges the goblin. He hits the rock floor. Aren strikes down at the goblin. The goblin tries to roll out of the way, but can't. The sword slashes his side. He stops moving.
> 
> Briana takes the spear. From the shadow of the caverns, another orc comes into view. He lifts his spear.
> 
> Briana kicks the orc's right shin. He grunts, and falls to the cavern floor. He tries to scramble out of reach. Aren strikes down with his sword at the orc. The orc tries to roll out of the way but can't. The sword cuts his side. He stops moving.

Here's the algorithm in action:

![nanogenmo2017](https://user-images.githubusercontent.com/919717/33417176-3a2b0626-d555-11e7-9f55-b8df40c68115.gif)

**Comment by hugovk** [comment]
"It's just a flesh wound!"

**Comment by filiph** [comment]
Some more context of the fight/NLG system employed here: https://medium.com/@filiph/skyrim-rendered-in-text-1899548ab2c4. In the article, I'm also explaining the concept of "fractal stories" employed in both the book and this NaNoGenMo entry.

I just realized you'll have trouble running the code because I based this hack on the master branch of Insignificant Little Vermin, which depends on a currently closed-source package. If you want to play around with the system, use the `dev` branch.

Since I'm reusing code for a game, there's a lot of it. If you want to read the interesting parts, here are a few tips:

* [Situation](https://github.com/filiph/edgehead/blob/nanogenmo2017/lib/fractal_stories/situation.dart#L18)
* [WorldState](https://github.com/filiph/edgehead/blob/nanogenmo2017/lib/fractal_stories/world.dart#L15)
* [Entity](https://github.com/filiph/edgehead/blob/nanogenmo2017/lib/fractal_stories/storyline/storyline_entity.dart#L3)
* [Storyline](https://github.com/filiph/edgehead/blob/nanogenmo2017/lib/fractal_stories/storyline/storyline.dart#L126) is the ancient, template-based NLG algorithm running the text generation

**Comment by filiph** [comment]
Ok, I couldn't resist and made some minor bug fixing today, plus added a script for PDF generation. There are now 2 outputs.

Thanks @hugovk for organizing this. It was a lot of fun!

**Comment by maetl** [comment]
This is great! I remember thinking when reading your article about fractal stories a few weeks ago that a lot of these ideas could be applied to generating books/NaNoGenMo texts as well—nice to see some amusing experiments based on that core work.

**Comment by superMDguy** [comment]
Yeah, this is incredible! I loved reading about and playing your "Insignificant Little Vermin" text adventure.

**Comment by hendrikboom3** [comment]
Briana is a real survivor, isn't she?
